stacks-core/src/utils.rs

```diff
@@ -111,6 +112,36 @@ impl From<StacksAddress> for PrincipalData {
 	}
 }
 
+impl TryFrom<String> for PrincipalData {
+	type Error = StacksError;
+
+	fn try_from(value: String) -> Result<Self, Self::Error> {
+		let parts: Vec<&str> = value.split('.').collect();
+
+		match parts.len() {
+			1 => {
+				let address = StacksAddress::try_from(parts[0])?;
+				Ok(Self::Standard(StandardPrincipalData::from(address)))
+			}
+			2 => {
+				let address = StacksAddress::try_from(parts[0])?;
+				let contract_name =
+					ContractName::new(parts[1]).map_err(|_err| {
+						StacksError::InvalidData("Invalid contract name")
```

Comment:
Error dilution. What 'contract' led to the error? What information is being lost from '_error.'?

Reply:
I don't know how to convert the error. Maybe we should have a new error type and new reuse StacksError

Reply:
Can I try?

Reply:
Thank you